pandas/core/internals.py

```diff
@@ -1963,6 +1963,23 @@ def concat_same_type(self, to_concat, placement=None):
         return self.make_block_same_class(values, ndim=self.ndim,
                                           placement=placement)
 
+    def fillna(self, value, limit=None, inplace=False, downcast=None,
```

Comment:
This was moved from Categorical.fillna with changes

1.) Removed check for limit, since that's done in `values.fillna`
2.) Removed `_try_coerce_result`
    For CategoricalBlock, this was unnecessary since
    `Categorical.fillna` always returns a Categorical
3.) Used `make_block_same_class`
    This limits `ExtensionArray.fillna` to not change the type
    of the array / block, which I think is a good thing.